src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyInputBuilderService.java

```diff
@@ -0,0 +1,514 @@
+                        .withValue("id", branch.getId())
+                        .build());
+                    throw new PowsyblException("Temporary limit '" + branches.getLimitNameNm1() + "' not found for branch '" + branch.getId() + "' !!");
+                }
+            }
+
+            if (!result.isEmpty()) {
+                context.getNonEvacuatedEnergyInputs().getBranchesThresholds().put(monitoredEquipment.getId(), monitoredBranchThreshold);
+            }
+        });
+
+        return result;
+    }
+
+    private void buildSensitivityFactorsAndVariableSets(NonEvacuatedEnergyRunContext context,
+                                                        Network network,
+                                                        Reporter reporter) {
+        List<NonEvacuatedEnergyMonitoredBranches> monitoredBranches = context.getNonEvacuatedEnergyInputData().getNonEvacuatedEnergyMonitoredBranches();
+        List<NonEvacuatedEnergyGeneratorLimitByType> generatorsLimitByType = context.getNonEvacuatedEnergyInputData().getNonEvacuatedEnergyGeneratorsLimit().getGenerators();
+
+        // build inputs for the sensitivities in MW per generation kind (similar to sensitivity analysis computation with injections set)
+        generatorsLimitByType.stream()
+            .filter(NonEvacuatedEnergyGeneratorLimitByType::isActivated) // we keep only activated generators limits
+            .forEach(generatorLimitByType -> {
+                // build sensitivity variable sets from generators limits in input data
+                List<SensitivityVariableSet> vInjectionsSets = buildSensitivityVariableSets(
+                    context.getNetworkUuid(), context.getVariantId(), network, reporter,
+                    List.of(IdentifiableType.GENERATOR),
+                    generatorLimitByType.getGenerators(),
+                    SensitivityAnalysisInputData.DistributionType.PROPORTIONAL_MAXP);
+                context.getNonEvacuatedEnergyInputs().addSensitivityVariableSets(vInjectionsSets);
+
+                // we store the cappings generators by energy source
+                context.getNonEvacuatedEnergyInputs().getCappingsGenerators().put(generatorLimitByType.getEnergySource(),
+                    vInjectionsSets.stream().flatMap(v -> v.getVariablesById().keySet().stream()).toList());
+
+                // build sensitivity factors from the variable sets (=set of generators), the contingencies and the monitored branches sets
+                monitoredBranches.stream()
+                    .filter(NonEvacuatedEnergyMonitoredBranches::isActivated)  // we keep only activated monitored branches set
+                    .forEach(branches -> {
+                        List<SensitivityFactor> fInjectionsSet = buildSensitivityFactorsFromVariablesSets(
+                            context, network, reporter,
+                            List.of(IdentifiableType.LINE, IdentifiableType.TWO_WINDINGS_TRANSFORMER),
```

Comment:
Can we put this restriction to Line and 2WT in a separate explicit entry that we can maintain easily?